commit 16dd10970fee8724b1a199a21517a6349ff8a51d
Author: Johan Hovold <johan@kernel.org>
Date:   Tue Jul 3 12:05:47 2018 +0200

    misc: sram: fix resource leaks in probe error path
    
    commit f294d00961d1d869ecffa60e280eeeee1ccf9a49 upstream.
    
    Make sure to disable clocks and deregister any exported partitions
    before returning on late probe errors.
    
    Note that since commit ee895ccdf776 ("misc: sram: fix enabled clock leak
    on error path"), partitions are deliberately exported before enabling
    the clock so we stick to that logic here. A follow up patch will address
    this.
    
    Cc: stable <stable@vger.kernel.org>     # 4.9
    Cc: Alexandre Belloni <alexandre.belloni@free-electrons.com>
    Signed-off-by: Johan Hovold <johan@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/misc/sram.c b/drivers/misc/sram.c
index fc0415771c00..4dd0d868ff88 100644
--- a/drivers/misc/sram.c
+++ b/drivers/misc/sram.c
@@ -407,13 +407,20 @@ static int sram_probe(struct platform_device *pdev)
 	if (init_func) {
 		ret = init_func();
 		if (ret)
-			return ret;
+			goto err_disable_clk;
 	}
 
 	dev_dbg(sram->dev, "SRAM pool: %zu KiB @ 0x%p\n",
 		gen_pool_size(sram->pool) / 1024, sram->virt_base);
 
 	return 0;
+
+err_disable_clk:
+	if (sram->clk)
+		clk_disable_unprepare(sram->clk);
+	sram_free_partitions(sram);
+
+	return ret;
 }
 
 static int sram_remove(struct platform_device *pdev)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1033_16dd10970fee8724b1a199a21517a6349ff8a51d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1033_16dd10970fee8724b1a199a21517a6349ff8a51d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
